tests/package_structure_test.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2023 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests of the JAX public package structure"""
+
+import importlib
+import types
+from typing import Sequence
+
+from absl.testing import absltest, parameterized
+
+from jax._src import test_util as jtu
+
+
+def _mod(module_name: str, *, include: Sequence[str] = (), exclude: Sequence[str] = ()):
+  return {"module_name": module_name, "include": include, "exclude": exclude}
+
```

Comment:
Good idea. I think we should also have a test that verifies the set of default-exported names (or at least modules) is exactly what we think it should be.

Not for this change.

Reply:
I'll keep that in mind for later.